File: trail1/조건에 따른 연산/operatino-by-rule.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int cnt = 0;
        while (N < 1000) {
            if (N % 2 == 0) {N = 3*N + 1;}
            else {N = 2*N + 2;}
            cnt++;
        }
        System.out.println(cnt);
    }
}

File: trail1/조건에 따른 연산/operatino-by-rule.cpp
#include <iostream>
using namespace std;

int main() {
    int N;
    cin >> N;
    int cnt = 0;
    while (N < 1000) {
        if (N % 2 == 0) {N = N * 3 + 1;}
        else {N = N * 2 + 2;}
        cnt++;
    }
    cout << cnt << endl;
    return 0;
}